sphinx/config.py

```diff
@@ -163,9 +166,23 @@ def __init__(self, config: Dict[str, Any] = {}, overrides: Dict[str, Any] = {})
 
     @classmethod
     def read(cls, confdir: str, overrides: Dict = None, tags: Tags = None) -> "Config":
-        """Create a Config object from configuration file."""
-        filename = path.join(confdir, CONFIG_FILENAME)
-        namespace = eval_config_file(filename, tags)
+        """Create a Config object from a configuration file."""
+        config_py = path.join(confdir, CONFIG_FILENAME_PY)
+        config_yaml = path.join(confdir, CONFIG_FILENAME_YAML)
+        if path.isfile(config_py):
+            if path.isfile(config_yaml):
+                raise ApplicationError(
```

Comment:
I noticed this error is related to the configuration. So it would be much better to use `ConfigError` instead.

Reply:
done